lib/ToolbarPanelHub.jsm

```diff
@@ -181,6 +182,29 @@ class _ToolbarPanelHub {
     });
   }
 
+  /**
+   * Attach click event listener defined in message payload
+   */
+  _attachClickListener(win, element, message) {
+    element.addEventListener("click", () => {
+      switch (message.content.layout) {
+        case "tracking-protections":
+          win.switchToTabHavingURI(message.content.cta_url, true);
+          break;
+        default:
+          win.ownerGlobal.openLinkIn(message.content.cta_url, "tabshifted", {
```

Comment:
You might want to consider exposing access to `handleUserAction` in ASRouter since it would give you access to opening about pages, about:preferences, etc.